### PR TITLE
add autoscaling permissions to the iam policy

### PR DESCRIPTION
**Pull request by teemow**
towards https://github.com/giantswarm/giantswarm/issues/4543

### CODE REVIEW

File: service/controller/v19/templates/cloudformation/guest/iam_policies.go [hunk 1]
```diff
@@ -114,6 +114,14 @@ const IAMPolicies = `{{define "iam_policies"}}
...
               - "ecr:BatchGetImage"
             Resource: "*"
+
+          - Effect: "Allow"
+            Action:
+              - "autoscaling:DescribeAutoScalingGroups"
+              - "autoscaling:DescribeAutoScalingInstances"
+              - "autoscaling:SetDesiredCapacity"
+              - "autoscaling:TerminateInstanceInAutoScalingGroup"
+            Resource: "*"
```
Comment (pipo02mix) [inline service/controller/v19/templates/cloudformation/guest/iam_policies.go]:
```suggestion
              - "autoscaling:autoscaling:DescribeTags"              
            Resource: "*"
            
          - Effect: "Allow"
            Action:
              - "autoscaling:SetDesiredCapacity"
              - "autoscaling:TerminateInstanceInAutoScalingGroup"
            Resource: "*"
            Condition: 
              StringEquals:   
                aws:RequestTag/giantswarm.io/cluster: "{{ $v.ClusterID }}" 
```
We still need to add the `ClusterID` in the iam policy adapter. I can do push a commit with that tomorrow
I would like to nail it down so only cluster ASG can be touched?

File: service/controller/v19/templates/cloudformation/guest/iam_policies.go [hunk 2]
```diff
@@ -114,6 +114,14 @@ const IAMPolicies = `{{define "iam_policies"}}
               - "ecr:ListImages"
               - "ecr:BatchGetImage"
             Resource: "*"
+
+          - Effect: "Allow"
+            Action:
+              - "autoscaling:DescribeAutoScalingGroups"
+              - "autoscaling:DescribeAutoScalingInstances"
+              - "autoscaling:SetDesiredCapacity"
+              - "autoscaling:TerminateInstanceInAutoScalingGroup"
```
Comment (pipo02mix) [inline service/controller/v19/templates/cloudformation/guest/iam_policies.go]:
```suggestion
```

File: service/controller/v19/templates/cloudformation/guest/iam_policies.go [hunk 3]
```diff
@@ -114,6 +114,14 @@ const IAMPolicies = `{{define "iam_policies"}}
               - "ecr:ListImages"
               - "ecr:BatchGetImage"
             Resource: "*"
+
+          - Effect: "Allow"
+            Action:
+              - "autoscaling:DescribeAutoScalingGroups"
+              - "autoscaling:DescribeAutoScalingInstances"
+              - "autoscaling:SetDesiredCapacity"
```
Comment (pipo02mix) [inline service/controller/v19/templates/cloudformation/guest/iam_policies.go]:
```suggestion
```

**Comment by pipo02mix** [comment]
Maybe it is better to add my changed to a new branch and PR to here. Suggestions looks weird. I would do tomorrow

**Comment by teemow** [comment]
Nice! I've added your changes.

**Comment by pipo02mix** [comment]
PR with missing part to his branch https://github.com/giantswarm/aws-operator/pull/1298/files

**Comment by teemow** [comment]
@pipo02mix thx. I've cherry-picked the commit. I'll run e2e tests now. So we can get this into `6.1.0`

**Pull request merged**